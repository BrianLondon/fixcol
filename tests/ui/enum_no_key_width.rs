use fixcol_derive::ReadFixed;

#[derive(ReadFixed)]
struct MyStruct {
    #[fixcol(width = 5)]
    my_int: u16
}

#[derive(ReadFixed)]
#[fixcol(ignore_others = true)]
enum Thing1 {
    #[fixcol(key = "one")]
    Thing1(#[fixcol(width = 5)] u16),
    #[fixcol(key = "two", embed = true)]
    Thing2(MyStruct),
}

#[derive(ReadFixed)]
enum Thing2 {
    #[fixcol(key = "one")]
    Thing1(#[fixcol(width = 5)] u16),
    #[fixcol(key = "two", embed = true)]
    Thing2(MyStruct),
}

#[derive(ReadFixed)]
#[fixcol(key_width = three)]
enum Thing3 {
    #[fixcol(key = "one")]
    Thing1(#[fixcol(width = 5)] u16),
    #[fixcol(key = "two", embed = true)]
    Thing2(MyStruct),
}

pub fn main() {}
