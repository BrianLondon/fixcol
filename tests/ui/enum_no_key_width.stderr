error: fixcol-derive error: The parameter 'key' must be provided for all enum variants.

        Try adding #[fixcol(key_width = 10)] to this enum replacing "10" with the width of your key.
  --> tests/ui/enum_no_key_width.rs:11:6
   |
11 | enum Thing1 {
   |      ^^^^^^

error: fixcol-derive error: The parameter 'key' must be provided for all enum variants.

        Try adding #[fixcol(key_width = 10)] to this enum replacing "10" with the width of your key.
  --> tests/ui/enum_no_key_width.rs:19:6
   |
19 | enum Thing2 {
   |      ^^^^^^

error: fixcol-derive error: Expected numeric value for key_width.
  --> tests/ui/enum_no_key_width.rs:27:22
   |
27 | #[fixcol(key_width = three)]
   |                      ^^^^^
